agent/src/main.rs

```diff
@@ -92,20 +88,37 @@ async fn main() -> Result<()> {
         .layer(DefaultBodyLimit::max(MAX_BODY_SIZE_BYTES))
         .with_state(state);
 
-    let addr = format!("{}:{}", host, port);
-    log::info!("serval agent daemon listening on {}", &addr);
+    let predefined_port: Option<u16> = match std::env::var("PORT") {
```

Comment:
I spent quite a bit of time trying to write this as a map, but Rust was too picky about the error types being compatible.

````rust
let predefined_port: Option<u16> = std::env::var("PORT")
    .and_then(|port_str| port_str.parse::<u16>())
    .ok();
````

👇  

```
error[E0308]: mismatched types
  --> agent/src/main.rs:96:30
   |
96 |         .and_then(|port_str| port_str.parse::<u16>())
   |                              ^^^^^^^^^^^^^^^^^^^^^^^ expected enum `VarError`, found struct `ParseIntError`
   |
   = note: expected enum `Result<_, VarError>`
              found enum `Result<u16, ParseIntError>`

```